funcx_sdk/funcx/utils/url_parsing.py

```diff
@@ -0,0 +1,32 @@
+from urllib.parse import urlparse
+
+from funcx.utils.errors import InvalidServiceAddress
+
+
+def validate_service_address(service_address):
+    try:
+        url_data = urlparse(service_address)
+        # port must be accessed to raise port value issues
+        url_data.port
+        if url_data.scheme != "http" and url_data.scheme != "https":
+            raise InvalidServiceAddress("Protocol must be HTTP/HTTPS")
+
+        if url_data.netloc is None or url_data.hostname is None:
+            raise InvalidServiceAddress("Address is malformed")
+    except Exception as e:
+        raise InvalidServiceAddress(f"Address is malformed - {e}")
+
+
+def ws_uri_from_service_address(service_address):
+    url_data = urlparse(service_address)
+    scheme = "wss" if url_data.scheme == "https" else "ws"
+    hostname = url_data.hostname
+    port = url_data.port
+    port_str = ""
+    # for testing purposes: the service address is typically
+    # localhost:5000 and ws uri is localhost:6000
+    if port == 5000:
```

Comment:
If for whatever reason, the service is hosted at `http://localhost:9000/v2`, this function will return `'ws://localhost:9000/ws/v2/', which isn't correct. If the port is specified and it isn't 5000 you probably should raise an error.

Reply:
@yadudoc I added this in the service address validation, but should we be concerned that anyone is developing by port forwarding with kubectl to a non-5000 port? If so, I suppose it should be discouraged anyway